eventhandler/src/main/java/eu/arrowhead/core/eventhandler/service/EventHandlerService.java

```
@@ -0,0 +1,320 @@
+		
+		final Set<Subscription> involvedSubscriptions = eventHandlerDBService.getInvolvedSubscriptions(request);
+		
+		filterInvolvedSubscriptionsBySubscriptionParameters(involvedSubscriptions, request);
+		
+		eventHandlerDriver.publishEvent(request, involvedSubscriptions);
+		
+		return new EventPublishResponseDTO(); //always return empty response
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	@Transactional
+	public void publishSubscriberAuthorizationUpdateRequest(final EventPublishRequestDTO request) {
+		logger.debug("publishSubscriberAuthorizationUpdateRequest started ...");
+		
+		validateAuthorizationUpdateEventType( request.getEventType() );
+		final Long subscriberSystemId = validateAuthorizationUpdatePayload( request.getPayload() );
+		
+		final List<Subscription> involvedSubscriptions = eventHandlerDBService.getInvolvedSubscriptionsBySubscriberSystemId( subscriberSystemId );
+		if ( involvedSubscriptions.isEmpty() ) {
+			
+			return;
+		}		
+
+		final SystemRequestDTO subscriber = DTOConverter.convertSystemToSystemRequestDTO( involvedSubscriptions.get(0).getSubscriberSystem() );		
+		final Set<SystemResponseDTO> authorizedPublishers = eventHandlerDriver.getAuthorizedPublishers(subscriber);
+		
+		eventHandlerDBService.updateSubscriberAuthorization( involvedSubscriptions, authorizedPublishers );
+		
+	}
+	
+	//-------------------------------------------------------------------------------------------------
+	@Transactional
+	public SubscriptionResponseDTO updateSubscriptionRequest(final long id, final SubscriptionRequestDTO subscriptionRequestDTO) {
+		logger.debug("updateSubscriptionRequest started ...");
+		
+		final Subscription updatedSubscription = updateSubscription( id, subscriptionRequestDTO );
+		
+		if ( updatedSubscription == null || Utilities.isEmpty(updatedSubscription.getNotifyUri())) {
+			
+			return new SubscriptionResponseDTO(); // return empty SubscriptionResponseDTO
```

updateSubscription should not return with null or allow empty notifyUri. Instead it should throw exception if the update failed or the request contains invalid parameter value.